### PR TITLE
Clair v4 `IndexState` Endpoint

### PR DESCRIPTION
**Pull request by alecmerdler**
Fixes https://issues.redhat.com/browse/PROJQUAY-610

### CODE REVIEW

**Comment by alecmerdler** [comment]
@thomasmckay This will need to be cherry-picked into `3.3-release`.

**Comment by ldelossa** [comment]
Should the Interface impl have it's method changed as well? 

https://github.com/quay/quay/pull/340/files#diff-de04d2aed57baa3c456f91913d059b04R47

**Comment by alecmerdler** [comment]
@ldelossa That's just the name of the method. Unless you are planning on adding `matcher_state`, then it would maybe make sense. The methods can't be 1:1 with the routes, for example `POST /index_report --> index()` and `GET /index_report --> `index_report()`.

File: util/secscan/v4/api.py [hunk 1]
```diff
@@ -204,5 +204,6 @@ def is_valid_response(action, resp={}):
         try:
             validate(resp, schema, resolver=resolver)
             return True
-        except Exception:
+        except Exception as e:
+            logger.exception(e)
```
Comment (josephschorr) [inline util/secscan/v4/api.py:208]: `logger.exception` will already add the exception contents; please put a descriptive message on this

Comment (alecmerdler) [reply util/secscan/v4/api.py]: Fixed

**Pull request merged**